Downloads and installs loki binary conditionally and on the target host

funkyfuture opened the pull request:
this commit achieves two things:

- it only downloads loki when the installed version is not the expected one
- it handles everything on the target host

it's quiet a while since i made this for my usage. probably because i use this role among others in a playbook and wanted to reduce their playtime. so the purpose is to reduce used resources.

funkyfuture committed:
Downloads and installs loki binary conditionally and on the target host

rlex commented:
I intentionally made it installing via local machine for cases when target system is airgapped and can't download loki binary from github.

funkyfuture commented:
i see. i have no opinion on it yet, but adding the capabilities for both variants seems to produce too much redundancy. do you have one?

funkyfuture committed:
Adds variable to configure the role's download tactic

defaults to download via localhost.
the binaries are extracted on the target host though to avoid duplicated 
"code".

funkyfuture commented:
i re-enabled that behaviour and added an option for direct downloads.
